src/api/test-controller.js

```diff
@@ -230,6 +232,27 @@ export default class TestController {
 
         return selector();
     }
+
+    _setNativeDialogHandler$ (fn, dependencies) {
+        var builder = new ClientFunctionBuilder(fn, { dependencies }, {
+            instantiation: 'setNativeDialogHandler',
+            execution:     'setNativeDialogHandler'
+        });
+
+        var compiledHandler     = builder.compiledFnCode;
+        var encodedDependencies = builder.replicator.encode(dependencies || {});
```

Comment:
Use `builder.getCommand` to achieve function descriptor. The use `ClientFunctionExecutor.getResult()` to achieve function result on client (since it will be executed multiple times, I gues it makes sense to extract function evaluation to ctor in `ClientFunctionExecutor`.